File: median/trivial.go
package median

import (
	"math"
	"sort"
)

func median(s []float64) float64 {
	sort.Float64s(s)
	n := len(s)
	if n%2 == 0 {
		return (s[n/2-1] + s[n/2]) / 2.
	}
	return s[n/2]
}

func RollingMedianTrivial(h int, x []float64) []float64 {
	n := len(x)
	res := make([]float64, 0, n)
	for i := 0; i < h; i++ {
		res = append(res, math.NaN())
	}
	for i := h; i < n-h; i++ {
		res = append(res, median(x[i-h:i+h+1]))
	}
	for i := 0; i < h; i++ {
		res = append(res, math.NaN())
	}
	return res
}

File: median/trivial.py
def median(arr):
	sarr = sorted(arr)
	n = len(arr)
	if n%2 == 0:
		return (sarr[n//2-1] + sarr[n//2]) // 2
	return sarr[n//2]
	
def rolling_median_trivial(h, arr):
	n = len(arr)
	res = []
	for i in range(h):
		res.append(float('nan'))
	for i in range(h, n-h):
		res.append(median(arr[i-h:i+h+1]))
	for i in range(h):
		res.append(float('nan'))
	return res
